adala/environments/kafka.py

```diff
@@ -67,8 +67,10 @@ async def initialize(self):
         await self.producer.start()
 
     async def finalize(self):
-        await self.consumer.stop()
-        await self.producer.stop()
+        if self.consumer:
```

Comment:
when would we not have a consumer or producer?

Reply:
If this function gets called before `initialize()`, as it might when the job is cancelled by the asyncio task group because the output job failed. It happened in testing